File: ts/src/basic/Classes.ts
class Circle {
    x: number;
    y: number;
    radius: number;

    constructor(x: number, y: number, radius: number) {
        this.x = x;
        this.y = y;
        this.radius = radius;
    }

    area(): number {
        return Math.PI * (this.radius * this.radius);
    }
}

export function run(): void {
    const c = new Circle(0.0, 0.0, 2.0);
    console.log(`circle ${c.x} ${c.y} ${c.radius}`);
}

File: rust/src/basic/classes.rs
/**
  * Created by sebby on 9/25/16.
  */

use std;

#[allow(dead_code)]
struct Circle {
    x: f64,
    y: f64,
    radius: f64,
}

#[allow(dead_code)]
impl Circle {
    fn new(x: f64, y: f64, radius: f64) -> Circle {
        Circle {
            x: x,
            y: y,
            radius: radius,
        }
    }

    fn area(&self) -> f64 {
        std::f64::consts::PI * (self.radius * self.radius)
    }
}

pub fn run() {
    let c = Circle::new(0.0, 0.0, 2.0);
    println!("circle {} {} {}", c.x, c.y, c.radius);
}

